SWATCH-3423: Migrate Gradle to Maven

Sgitario opened the pull request:
Jira issue: SWATCH-3423

## Description

After the analysis in https://docs.google.com/document/d/1NzEOq2phDtGnNm7nnPdZhEkF3jLjDy2L0wVQkbctf9Q/edit?tab=t.0,
All the work to migrate from Gradle to Maven has already been done in https://github.com/RedHatInsights/rhsm-subscriptions/pull/4288.
The only components that need to be done in a follow-up task are:

- Fix the code coverage analysis: at the moment, it includes generated classes which is doing the code coverage percentage is very slow.
- Migrate the Sonar plugin

## Notes
This section aims with how to easily start by testing these changes in a developer perspective.

1.- Start with command line by running: `./mvnw clean install`

This command will build, and install the artifacts into your local. Also, it will execute all the tests locally.

2.- In Intellij, don't use the same project because IntelliJ is configured to use Gradle which after these changes is gone.
Instead, create a new project from existing sources and this time, IntelliJ will recognise the project as Maven.

3.- In Intellij, enable the delegate IDE build/run actions to Maven:

![Screenshot From 2025-03-31 07-38-04](https://github.com/user-attachments/assets/bbc2560e-38c8-46d2-8056-7f7bc7cf00d0)

After this setting, you will be able to run tests and services from Intellij.

4.- How to run one service in Dev mode from the root folder.

For Quarkus services, example: `./mvnw -pl swatch-metrics -DskipTests quarkus:dev`
For Spring Boot services, example: `./mvnw -pl swatch-tally -DskipTests spring-boot:run`
Append all the properties using `-DpropertyName=value`. Example: `-DSERVER_PORT=8001`.

## Gradle vs Maven – Command Line Equivalents

| **Gradle Command**                                           | **Maven Command**                                                                                      | **Description**                                                  |
|--------------------------------------------------------------|---------------------------------------------------------------------------------------------------------|------------------------------------------------------------------|
| `./gradlew clean`                                            | `./mvnw clean`                                                                                          | Deletes the build directory (`build/` or `target/`).             |
| `./gradlew build`                                            | `./mvnw package`                                                                                        | Compiles the project and creates the artifact (`.jar`, etc.).    |
| `./gradlew test`                                             | `./mvnw test`                                                                                           | Runs unit tests.<br>Add `-DskipTests` to skip all the tests.     |
| `./gradlew install`                                          | `./mvnw install`                                                                                        | Installs the artifact to the local Maven repository.             |
| `./gradlew assemble`                                         | `./mvnw compile`                                                                                        | Compiles source code only.                                       |
| `./gradlew dependencies`                                     | `./mvnw dependency:tree`                                                                               | Displays the dependency tree.                                    |
| `./gradlew :module-rest:bootRun`                             | `./mvnw -pl module-rest spring-boot:run`                                                               | Runs the `module-rest` Spring Boot application.<br>Run `./mvnw install` beforehand. |
| `./gradlew :module-rest:quarkusDev`                          | `./mvnw -pl module-rest quarkus:dev`                                                                   | Runs the `module-rest` Quarkus app in dev mode.<br>Run `./mvnw install` beforehand. |
| `./gradlew tasks`                                            | `mvn help:describe -Dplugin=org.apache.maven.plugins:maven-help-plugin -Ddetail`                       | Lists available tasks and plugins.                               |
| `./gradlew spotlessCheck checkstyleMain checkstyleTest`     | `./mvnw package validate checkstyle:checkstyle -DskipTests`                                            | Validate the format.                                             |
| `./gradlew :swatch-database:run`                             | `./mvnw -f swatch-database/pom.xml exec:java`                                                          | Run database migrations.                                         |

## Tests
Regression testing. 

No gradle is mentioned anylonger in our repository.

InsightsDroid commented:
### IQE Tests Summary Report
 226 passed, 2 skipped, 234 deselected, 301 warnings in 2027.13s (0:33:47) 
Result: Success
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions

InsightsDroid commented:
### IQE Tests Summary Report
 229 passed, 2 skipped, 234 deselected, 304 warnings in 2011.63s (0:33:31) 
Result: Success
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions

Sgitario commented:
@lindseyburnett I reverted your commit because it's not necessary to copy the mvnw wrapper to all the subfolders.
Also, updated the PR to run the Quarkus/Spring Boot services from the root folder (updated the PR description as well)

InsightsDroid commented:
### IQE Tests Summary Report
 229 passed, 2 skipped, 235 deselected, 304 warnings in 2031.91s (0:33:51) 
Result: Success
IQE plugin image: https://quay.io/cloudservices/iqe-tests:rhsm-subscriptions

awood commented on config/checkstyle/checkstyle-xpath-suppressions.xml:
```diff
@@ -1,45 +0,0 @@
-<?xml version="1.0"?>
```
Why is this file removed?  The reason it exists is so that we do not use a custom checkstyle.xml (See #3983's notes).  The goal is to use the stock checkstyle.xml that we can just update over time.  Checkstyle does make breaking DTD changes and having to navigate those is a pain.  It's easier just to drop in the published Google checkstyle.xml file every time.  Any customizations we make are stored in the suppressions file(s) which don't change format that often.
